clean.js

```diff
@@ -0,0 +1,165 @@
+
+// setting up the sqs services
+const notificationQueueUrl =
+  "https://sqs.eu-central-1.amazonaws.com/816069166828/NotificationQueue";
+const actionBuilderURL =
+  "https://sqs.eu-central-1.amazonaws.com/816069166828/action-builder-q";
+
+// initializing variables
+let username;
+let recipients;
+let transaction_number;
+let orderID;
+let actionID;
+let vaultID;
+let tgUsername;
+let amount;
+let userId;
+
+// function to delete message from the action builder queue
+async function deleteMessage(receiptHandle) {
+  const deleteParams = {
+    QueueUrl: actionBuilderURL,
+    ReceiptHandle: receiptHandle,
+  };
+
+  try {
+    await sqs.deleteMessage(deleteParams).promise();
+    console.log("Message deleted successfully: ", receiptHandle);
+  } catch (error) {
+    console.error("Error deleting message:", error);
+  }
+}
+
+// function to get chat id from the username mentioned in the order
+async function getChatIdFromUsername(username) {
+  const tgResponse = await axios.get(telegramDb);
+  const tgUsersData = tgResponse.data;
+
+  for (const user of tgUsersData) {
+    if (user.telegram_user === username) {
+      return user.session_id;
+    }
+  }
+
+  return Response.json({ message: "session id not found" });
+}
+
+// function to send message to the user
+async function sendMessageToUser(username, message) {
+  try {
+    const chatId = await getChatIdFromUsername(username);
+    if (chatId) {
+      await bot.sendMessage(chatId, message);
+    }
+  } catch (error) {
+    console.log("error: ", error);
+  }
+}
+
+// function to post data to the telegram db
+async function postDataToTgDb(username, chatId) {
+  const postData = {
+    telegram_user: username,
+    session_id: chatId,
+  };
+
+  axios
+    .post(telegramDb, postData, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+    .then((response) => {
+      console.log("Success: ", response);
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+    });
+}
+// function to handle telegram updates from the user chats
+async function handleTelegramUpdate(body) {
+  console.log("body: ", body);
+  const update = JSON.parse(body);
+  await bot.processUpdate(update);
+
+  // handling specific bot commands
+  if (update.message && update.message.text === "/start") {
+    const chatId = update.message.chat.id;
+    const username = update.message.chat.username;
+
+    const response = await axios.get(dbUrl);
+    const data = response.data;
+    for (item of data) {
+      if (item.session_id === chatId) {
+        console.log("session ID already present in json");
+      } else {
+        await postDataToTgDb(username, chatId);
+      }
+    }
+    await bot.sendMessage(chatId, `Bot has started`);
+    return;
+  }
+
+  return {
+    statusCode: 200,
+    body: JSON.stringify({ message: "Telegram update processed successfully" }),
+  };
+}
+
+// main lambda handler
+exports.handler = async (event) => {
```

Comment:
I think it would be more clean to further refactor this, something like the following but please review and verify before adding in:

```
const processRecord = async (record) => {
  const {
    Recipients: recipients,
    Tg_Username: tgUsername,
    Amount: amount,
    User_ID: userId,
    Transaction_Index: transactionIndex,
    Order_id: orderId,
    Vault_ID: vaultId,
    Action_ID: actionId
  } = JSON.parse(record.body);

  console.log("Parsed record:", { recipients, tgUsername, amount, userId, transactionIndex, orderId, vaultId, actionId });

  // Delete SQS message early to prevent duplicate processing
  await deleteMessage(record.receiptHandle);
  console.log("Message deleted:", record.messageId);

  // Construct blink URL
  const blinkUrl = `https://dial.to/?action=solana-action%3Ahttps://squint-api.vercel.app/actions/${actionId}`;

  // Determine and send appropriate message
  if (recipients) {
    await sendReviewMessage(recipients, transactionIndex, blinkUrl);
  } else if (tgUsername) {
    await sendRequestMessage(tgUsername, amount, blinkUrl);
  }
};

const sendReviewMessage = async (recipients, transactionIndex, blinkUrl) => {
  const message = [
    'crct sent you a blink for reviewing',
    `Transaction number: ${transactionIndex}`,
    blinkUrl
  ].join('\n');

  await sendMessageToUser(recipients, message);
};

const sendRequestMessage = async (tgUsername, amount, blinkUrl) => {
  const message = [
    `${tgUsername} has requested ${amount} usdc from you`,
    blinkUrl
  ].join('\n');

  await sendMessageToUser(tgUsername, message);
};

exports.handler = async (event) => {
  console.log("Received event:", JSON.stringify(event, null, 2));

  try {
    if (event.Records) {
      await Promise.all(event.Records.map(processRecord));
    } else if (event.body) {
      return handleTelegramUpdate(event.body);
    }

    return {
      statusCode: 200,
      body: JSON.stringify({ message: "Message sent successfully" })
    };
  } catch (error) {
    console.error("Error processing event:", error);
    return {
      statusCode: 500,
      body: JSON.stringify({ error: "Failed to process message" })
    };
  }
};
```